packages/Search/src/details/DTK_DetailsTreeConstruction_def.hpp

```diff
@@ -254,32 +255,68 @@ void TreeConstruction<DeviceType>::sortObjects(
             morton_codes ),
         reducer );
     if ( result.min_val == result.max_val )
-        return;
+    {
+        Kokkos::View<size_t *, DeviceType> permute(
+            Kokkos::ViewAllocateWithoutInitializing( "permute" ), n );
+        iota( permute );
+        return permute;
+    }
     Kokkos::BinSort<Kokkos::View<unsigned int *, DeviceType>, CompType>
         bin_sort( morton_codes,
                   CompType( n / 2, result.min_val, result.max_val ), true );
     bin_sort.create_permute_vector();
     bin_sort.sort( morton_codes );
-    // TODO: We might be able to just use `bin_sort.get_permute_vector()`
-    // instead of initializing the indices with iota() and sorting the vector
-    bin_sort.sort( object_ids );
     Kokkos::fence();
+
+    // Kokkos::BinSort::get_permute_vector() returns a view with a
+    // value type set to the size type of the memory space.  Cuda memory space
+    // size type is `unsigned int` while it is `size_t` on the host...
+    auto permute = bin_sort.get_permute_vector();
+
+    // The static assertion below is meant to check whether changes occured in
+    // Kokkos_Sort.hpp or in the definition of the size types in which case we
+    // might want to reconsider what to do with the permutation indices.
+    static_assert(
+#ifdef KOKKOS_ENABLE_CUDA
+        std::is_same<typename decltype( permute )::traits::execution_space,
```

Comment:
Are you worried there will be changes to this in the future?

Reply:
I just want to know it if something changes.